view/frontend/web/js/track-checkout.js

```diff
@@ -0,0 +1,70 @@
+/* Track checkout funnel transition in magento's default checkout */
+
+// TODO: support logged in user
+
+var currentStep = '';
+
+function waitFor(condition, doWhenReady) {
+    var checkCondition = function () {
+        if (condition()) {
+            doWhenReady();
+        } else {
+            setTimeout(condition, 50);
```

Comment:
this will just call condition in 50ms? you want to call `checkCondition`?

Reply:
woops fixed thank you